manifests/init.pp

```diff
@@ -18,6 +18,12 @@
 # $host_private_key:: path to host private key, usually puppet agent private key
 #                     May be overriden if $::rh_certificate_consumer_host_key (from Facter) is found
 #
+# $foreman_repo_rel:: add / manage foreman-plugins yum repo and set to release version. Eg  '1.14'
+#                     type: string
+#
+# $foreman_rpm_key::  RPM Key source file for foreman-plugins repo. Note: Currently, packages are not signed
+#                     type: string
```

Comment:
I looked twice: This breaks puppet3 compat, as http sources appeared in [Puppet 4.4](https://tickets.puppetlabs.com/browse/PUP-1072). I would check for a source and if none, skip the file resource.